docs/proposals/volume-unification.md

```diff
@@ -0,0 +1,65 @@
+## Abstract
+
+A proposal for enabling several Kubernetes resources to be available
+within the same volume mount.
+
+## Motivation
+
+
+## Constraints and Assumptions
+
+It is assumed that any specified paths do not cause overlap with other
+resources. If a conflict does occur, it is considered an error.
+
+## Current State Overview
+
+## Use Cases
+
+1.  As a user, I want to automatically populate a single volume with the keys
+    from multiple secrets, configmaps, and with downward API information, so
+    that I can synthesize a single directory with various sources of
+    information.
+
+2.  As a user, I want to populate a single volume with the keys from multiple
+    secrets, configmaps, and with downward API information, explicitly
+    specifying paths for each item, so that I can have full control over the
+    contents of that volume.
+
+## Analysis
+
+Use Case: ConfigMap + Secrets + Downward API
```

Comment:
These aren't really use-cases so much as variations of the 2 main use-cases, so I would remove the 'use case' phrasing here.  I would also add an introduction explaining that there are several variations of the use-case that need to be thought through before you launch into this.  

Basically, a proposal is the story of what the problem is you want to solve, what state it's in now, an in-depth analysis of the problem space, and then finally specifics of how you intend to solve it.